File: solute/epfl/components/buttonradio/buttonradio.py
from solute.epfl.components.form.inputbase import FormInputBase


class ButtonRadio(FormInputBase):
    """
    A form radio group using buttons as radio fields.

    Typically, this component is used in a form:

    .. code:: python

        form = Form(node_list=[ButtonRadio(label="Gender:", name="gender", default="male", options=["male", "female"])])

    """
    template_name = "buttonradio/buttonradio.html"
    compo_state = FormInputBase.compo_state + ['options']

    js_parts = FormInputBase.js_parts + ['buttonradio/buttonradio.js']
    js_name = FormInputBase.js_name + [("solute.epfl.components:buttonradio/static", "buttonradio.js")]
    css_name = FormInputBase.css_name + [("solute.epfl.components:buttonradio/static", "buttonradio.css")]

    options = ""  #: List of strings or key, value tuples presented as options.

    validation_type = 'text'  #: Evaluate this component as text.

    def __init__(self, page, cid, options=None, **extra_params):
        """A component displaying a radio form input with buttons.

        :param options: List of strings or key, value tuples presented as options.
        """
        super(ButtonRadio, self).__init__(page, cid, options=options, **extra_params)

File: solute/epfl/components/buttonradio/static/buttonradio.js
epfl.ButtonRadio = function (cid, params) {
    epfl.ComponentBase.call(this, cid, params);

    var selector = "input[type=radio][name=" + cid + "]";
    var compo = this;
    var enqueue_event = !params["fire_change_immediately"];

    $(selector).change(function(){
        epfl.FormInputBase.on_change(compo, $(this).val(), cid, enqueue_event);
    });

};

epfl.ButtonRadio.inherits_from(epfl.ComponentBase);
